### ISSUE TITLE
No support for Components anymore in 0.3.6?

### ISSUE THREAD

**Issue by Daveiano** [treyhuffine/lightbox-react#11]
I am using the Lightbox v 0.3.6 and when using React Components as Lightbox Sources it says:

```
warning.js:33 Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: object.

Check the render method of `ArtistDetail`.
    in ArtistDetail (created by I18n)
    in I18n (created by Translate(ArtistDetail))
    in Translate(ArtistDetail) (created by Query)
    in Query (created by Apollo(Translate(ArtistDetail)))
    in Apollo(Translate(ArtistDetail)) (created by Query)
    in Query (created by Apollo(Apollo(Translate(ArtistDetail))))
    in Apollo(Apollo(Translate(ArtistDetail))) (created by Route)
    in Route (created by App)
    in Switch (created by App)
    in Provider (created by App)
    in main (created by App)
    in App (created by Route)
    in Route (created by withRouter(App))
    in withRouter(App)
    in I18nextProvider
    in Router (created by BrowserRouter)
    in BrowserRouter
    in Provider
    in ApolloProvider
```
Code:

```
const lightBoxSources = props.artistDetail.route.entity.fieldWannaDoS.map((item, index) =>
      <Overlay
        img={item.entity.fieldMediaImage.image}
        heading={props.artistDetail.route.entity.title}
        url={props.match.url}
        themes={item.entity.fieldStyle}
        index={item.entity.fieldMediaImage.targetId}
        key={index}
        language={props.location.pathname.split('/')[1]}
        lightBoxType={'wannados'}
        close={this.closeOverlayWannaDos}
      />
    );
```

**Comment by treyhuffine** [comment]
Hi Daveiano, they should still be working. The error message says it expects a function/class which should be both types of React components.


Does the lightbox still correctly load the component in the UI?

Did you change any code other than upgrading the lightbox?

**Comment by treyhuffine** [comment]
I should probably make this more clear, but are you importing the case separately?

**Comment by Daveiano** [comment]
Sorry, but what do you mean by importing the case?

**Comment by treyhuffine** [comment]
Sorry autocorrect. CSS file.

import 'lightbox-react/style.css';

**Comment by Daveiano** [comment]
No I don't, but how should this solve the JS Error?

I am getting a blank page because of the errors, the whole app crashes and the react mounting is empty after that.

**Comment by Daveiano** [comment]
This is my Component I am giving to the Lightbox:

```
class Overlay extends Component{
  state = { marginRight: 0 }

  componentDidMount() {
    window.addEventListener('resize', this.resizeHandler);

    const { width, height, x, y } = contain(window.innerWidth, window.innerHeight, this.props.img.width, this.props.img.height);

    this.setState({ marginRight: x - 8 });
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.resizeHandler);
  }

  resizeHandler = () => {
    const { width, height, x, y } = contain(window.innerWidth, window.innerHeight, this.props.img.width, this.props.img.height);

    this.setState({ marginRight: x - 8 });
  }

  render() {
    console.log(this);
    const shareUrl = `${self}${this.props.url}?${this.props.lightBoxType}=${this.props.index}`;

    return (
      <div className="overlay-item">
        <img src={this.props.img.url} alt="alt" />
        <div className="overlay-wrap">
          <div className="overlay" style={{ right: `${this.state.marginRight}px` }}>
            <h1>
              {this.props.headingLink ? (
                <Link onClick={this.props.close} to={this.props.headingLink}>{this.props.heading}</Link>
              ) : (
                <span>{this.props.heading}</span>
              )}
            </h1>
            {this.props.themes.length > 0 &&
            <div className="themes">
              <ul>{this.props.themes.map((theme, index) =>
                <li key={index}>
                  <Link
                    onClick={this.props.close}
                    to={`/${this.props.language}${themeDetailPath}/${encodeURI(theme.entity.name.replace(/&/g, '-').replace(/ /g, '')).toLowerCase()}`}>
                    {theme.entity.name}
                  </Link>
                </li>
              )}
              </ul>
            </div>
            }
            <Link className="appointment-link" onClick={this.props.close} to={`/${this.props.language}${appointmentPath}`}>Termin sichern</Link>
            <div className="share">
              <div className="buttons">
                <FacebookShareButton
                  url={shareUrl}
                  quote={this.props.heading}
                  className="fb-share-button">
                  <FacebookIcon
                    size={32}
                    round />
                </FacebookShareButton>
                <PinterestShareButton
                  media={this.props.img.url}
                  url={shareUrl}
                  description={this.props.heading}
                  className="pinterest-share-button">
                  <PinterestIcon
                    size={32}
                    round />
                </PinterestShareButton>
                <div className="button copy-to-clipboard" title="Link in die Zwischenablage kopieren" onClick={() => copy(shareUrl)}>
                  <img src="/img/copy-to-clipboard.svg" />
                </div>
              </div>
              <div className="share-label">teilen</div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Overlay;
```

**Comment by Daveiano** [comment]
Some news:

I Imported Lightbox like this (because I think there was a Problem with SSR in the previous version)

```
let Lightbox;

if (typeof window !== 'undefined') {
  Lightbox = require('lightbox-react');
}
```

I changed this to the following:

```
import Lightbox from 'lightbox-react';
```

Not the Lightbox opens correctly, but I am getting another Error:

`warning.js:33 Warning: Failed prop type: Invalid prop mainSrc supplied to ReactImageLightbox.`

**Comment by treyhuffine** [comment]
What's surprising is it's a warning which typically doesn't crash applications.

**Comment by treyhuffine** [comment]
I'm actually running it locally and seeing this warning as well. It's just an issue with the PropType checking but it shouldn't impact rendering. The Lightbox itself is still working for me.

<img width="1276" alt="screen shot 2018-10-27 at 8 37 55 am" src="https://user-images.githubusercontent.com/11709986/47604163-ac750380-d9c3-11e8-80df-b448e096ffcb.png">

**Comment by Daveiano** [comment]
Yes same behavior in my application, everything works , but 3 warnings for nextSrx, prevSrc and mainSrc

**Comment by treyhuffine** [comment]
Got it, I'll take a look at the prop-types early next week and publish a new version. Thanks for reporting the issue @Daveiano

**Comment by Daveiano** [comment]
Thank you for your fast response and helpfulness!

**Comment by treyhuffine** [comment]
No problem! PropTypes should be fixed now in 0.3.7. Let me know if you run into any issues.

fixed in https://github.com/treyhuffine/lightbox-react/commit/1641f0b12725b025b286c9f5b6ffa57c7cb65c68
